src/articles/articles.controller.ts

```diff
@@ -23,22 +29,37 @@ export class ArticlesController {
   ) {}
 
   @Post('create')
+  @UseGuards(AuthGuard)
+  @UseGuards(ArticlesGuard)
+  @UseInterceptors(ClassSerializerInterceptor)
+  @ApiHeader({
+    name: 'Authorization',
+    description: 'Bearer token',
+  })
   @ApiResponse({ status: 201, description: 'New article created.', type: Article})
   @ApiResponse({ status: 400, description: 'Bad Request.'})
-  async createArticle(@Body() articleDto: ArticleDto): Promise<Article> {
-    return this.articlesService.createArticle(articleDto);
+  async createArticle(
+    @Body() articleDto: ArticleDto,
+    @Headers('authorization') authorization
+  ): Promise<Article> {
+    const [, token] = authorization.split(' ');
```

Comment:
Actually, a **token** or even **user** could be added to the request in AuthGuard,  and in controller, it could receive from request via @Request decorator. Not via @Headers.